wagtail/test/snippets/models.py

```diff
@@ -63,10 +63,7 @@ def __str__(self):
         return self.text
 
 
-@register_snippet
 class FilterableSnippet(index.Indexed, models.Model):
```

Comment:
This model should probably be moved to `wagtail.test.testapp.models` so this whole thing is more cohesive...?

Reply:
Any reason not to add a ``views.py`` in ``wagtail/tests/snippets``?

Reply:
I am curious what the wider context for dropping `@register_snippet` in favour of the `register_snippet()` in wagtail_hooks is. This is non-blocking, or anything like that

Reply:
@kaedroho There isn't a `wagtail/tests/snippets` directory yet. If you meant `wagtail/snippets/tests`, well I just thought that since we want to add the `register_snippet()` in a `wagtail_hooks.py` file, it's best to put the viewset in the `views.py` adjacent to the `wagtail_hooks.py`, which is in `wagtail.test.testapp`.

@zerolab We don't want to encourage developers to write references to views/viewsets in `models.py`. In addition to helping avoid circular imports, this approach is also closer to how registration works for `ModelAdmin`. Snippets are also Wagtail-specific, so I think it makes sense for `register_snippet()` to be in `wagtail_hooks.py`.

Note that `@register_snippet` still works if you don't need any further customisations.

Reply:
Thank you @laymonage. Makes perfect sense.

Reply:
@laymonage I think @kaedroho meant `wagtail/test/snipppets`. For some reason lost in time, our test suite has a distinct `wagtail.test.snippets` app in addition to the `wagtail.test.testapp` app where most of our test models live - figuring out the reason for that wouldn't be a bad thing, but I think it's out of scope for this PR :-) So since we're editing a model that exists within the `wagtail.test.snippets` app (at least for now...), it would make sense for the corresponding viewset to be defined at `wagtail.test.snippets.views`.